src/client/providers/view.ts

```diff
@@ -0,0 +1,95 @@
+import { combineLatest, Observable } from 'rxjs'
+import { catchError, map, switchMap } from 'rxjs/operators'
+import { ContributableViewContainer } from '../../protocol'
+import * as plain from '../../protocol/plainTypes'
+import { Entry, FeatureProviderRegistry } from './registry'
+
+export interface ViewProviderRegistrationOptions {
+    id: string
+    container: ContributableViewContainer
+}
+
+export type ProvideViewSignature = Observable<plain.PanelView>
+
+/** Provides views from all extensions. */
+export class ViewProviderRegistry extends FeatureProviderRegistry<
+    ViewProviderRegistrationOptions,
+    ProvideViewSignature
+> {
+    /**
+     * Returns an observable that emits the specified view whenever it or the set of registered view providers
+     * changes. If the provider emits an error, the returned observable also emits an error (and completes).
+     */
+    public getView(id: string): Observable<plain.PanelView | null> {
+        return getView(this.entries, id)
+    }
+
+    /**
+     * Returns an observable that emits all views whenever the set of registered view providers or their properties
```

Comment:
Re: https://github.com/sourcegraph/sourcegraph-extension-api/issues/105

To avoid introducing that same bug in this new API, I documented how provider errors are handled in `getViews` and `getView` and suppressed them in `getViews`.

cc @slimsag @chrismwendt @felixfbecker